pages/join/01-you.tsx

```diff
@@ -18,13 +23,57 @@ export async function getStaticProps() {
   };
 }
 
+export function checkIfLoggedIn(
+  router: any,
+  setSignInProps: (signInProps: SignInProps) => void,
+  setIsLoggedIn: (isLoggedIn: boolean) => void,
+  setName?: (name: string) => void,
+  setEmail?: (email: string) => void,
+  setLinkedInPicture?: (picture: string) => void,
+  setUserId?: (userId: string) => void
+) {
+  if (
+    typeof sessionStorage === "undefined" ||
+    sessionStorage.getItem(SessionStorageEnum.USER_NAME) === null ||
+    sessionStorage.getItem(SessionStorageEnum.USER_ID) === null
+  ) {
+    sessionStorage.setItem(SessionStorageEnum.PREVIOUS_PAGE, "/join/01-you");
+    router.push({ pathname: `/login` });
+  } else {
+    setName &&
+      setName(
+        capitalizeFirstLetters(
+          sessionStorage.getItem(SessionStorageEnum.USER_NAME)
+        )
+      );
+    setEmail && setEmail(sessionStorage.getItem(SessionStorageEnum.USER_EMAIL));
```

Comment:
need to add a conditional here based on feedback from @taylorkmho, we should only allow LinkedIn emails if the email is verified

Reply:
added conditional for Google but not for LinkedIn. Unfortunately the API call for `/v2/userinfo` in LinkedIn is the endpoint needed to indicate whether the user has verified their email, but with the current permissions we have for the LinkedIn API, we're unable to access that endpoint

However, I was able to confirm (via testing it out) that LinkedIn accounts cannot be created without the user verifying their email, so the assumption can be made that LinkedIn accounts have a verified email